### DISCUSSION TITLE
How to edit language config of plain "text"?

### DISCUSSION THREAD

**Discussion by Abdillah** [helix-editor/helix#7567] [Q&A]
I want to change default indentation of language not currently supported (hence detected as text), to 4 spaces.
I tried the following to no avail,

```
[[language]]
name = "text"
scope = "source.text"
roots = []
file-types = [ '*' ]
indent = { tab-width = 4, unit = "    " }
```

**Comment by gabydd** [comment]
There is no fallback language config, but you could instead add a new entry into your languages.toml for the language you are working with, and if there is a tree-sitter grammar or lsp you can also make a pr to this repo to upstream language support.

**Reply by Abdillah** [reply]
Okay.
That led me to think about scratch pad `[scratch]` and how to change the default indentation of it.

**Reply by the-mikedavis** [reply]
Scratch bufffer indentation can't currently be configured

**Reply by Abdillah** [reply]
Is there any reason not to implement a fallback language (i.e. plain text without file extension match)?
I can try to experiment with it if it is good on the principle and technical side of helix.

**Reply by the-mikedavis** [reply]
Scratch buffers and files where the language can't be detected have no language config associated with them. It would be a little awkward for scratch buffers to have a language config block because language servers would not work with them. For example you might configure ltex-ls for text but language servers require files to have paths. It wouldn't be a very small change to the codebase so I'm not sure it's worth it: the usecase seems sort of niche to me and you can work around it by creating a language block for "text" and using `:lang text` in buffers where you want those rules to apply

**Reply by Abdillah** [reply]
I see.. thank you for the insight

**Comment by kj** [comment]
This is honestly my most wanted Helix feature. I just want to be able to change the default tab indentation to 8 for files without a language!

**Reply by Keshyu** [reply]
Agreed. I like sketching ideas in helix. If only there was folding and I could set the indentation to 2... ;-;

**Comment by whoizit** [comment]
this works for me:

```
❯ cat .config/helix/languages.toml 
[[language]]
name = "default"
scope = "source.default"
comment-token = "#"
indent = { tab-width = 2, unit = "  " }
file-types = ["*"]
roots = []
auto-format = false
```
```
❯ cat .config/helix/config.toml
theme = "ayu_evolve"

[keys.normal]
"C-." = ":lang default"
```
